### PR TITLE
Git pre push hook

### PR DESCRIPTION
**Pull request by vandot**
Add git pre-push hook so that before pushing all tests are passed on local dev machine
Install it with `make hooks`, it will create symlink `.githooks/pre-push.bash -> .git/hooks/pre-push`

### CODE REVIEW

File: .githooks/pre-push.bash [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+if ! command -v golangci-lint &> /dev/null; then
+    echo "installing golangci-lint..."
+    go get -u github.com/golangci/golangci-lint/cmd/golangci-lint
+fi
+
+make lint
+
+make vet
+
+make test-race
```
Comment (janos) [inline .githooks/pre-push.bash:20]: Is it ok to combine this into `make lint vet test-race`?

File: Makefile [hunk 2]
```diff
@@ -39,6 +39,10 @@ build: export CGO_ENABLED=0
 build:
 	$(GO) build -trimpath -ldflags "$(LDFLAGS)" ./...
 
+.PHONY: hooks
```
Comment (janos) [inline Makefile:42]: I would call this rule `githooks`.

File: .githooks/pre-push.bash [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+exit 0
```
Comment (janos) [inline .githooks/pre-push.bash:22]: `exit 0` should be implicit, just curious why it is needed?

File: .githooks/pre-push.bash (hunk 1 above)
Comment (vandot) [reply .githooks/pre-push.bash:20]: @janos didn't tried, I will test if make will exit when first commands exits with error

File: .githooks/pre-push.bash (hunk 3 above)
Comment (vandot) [reply .githooks/pre-push.bash:22]: probably not, I like it like that :)
will check without it

**Comment by vandot** [comment]
@janos applied all suggested changes

**Pull request merged**